blockstore/splitstore/splitstore_compact.go

```diff
@@ -1222,8 +1222,10 @@ func (s *SplitStore) walkObjectIncomplete(c cid.Cid, visitor ObjectVisitor, f, m
 	}
 
 	// check this before recursing
-	if err := s.checkYield(); err != nil {
-		return sz, err
+	if yielding {
+		if err := s.checkYield(); err != nil {
```

Comment:
Another approach that is a bit cleaner is to change this (and maybe `walkObject`) to always call `checkClosing`, then do a `checkYield` in every call to [`walkBlock`](https://github.com/filecoin-project/lotus/pull/10855/files#diff-f39742667127f7440576464ad7286c341abd00ec405df1612a8bcceb6b8881abR976).  Its probably fine for sync protection to just do yielding once for every block walked during compaction.